packages/material-ui/src/List/List.js

```diff
@@ -43,24 +40,24 @@ const List = React.forwardRef(function List(props, ref) {
   const context = React.useMemo(() => ({ dense }), [dense]);
 
   return (
-    <Component
-      className={clsx(
-        classes.root,
-        {
-          [classes.dense]: dense && !disablePadding,
-          [classes.padding]: !disablePadding,
-          [classes.subheader]: subheader,
-        },
-        className,
-      )}
-      ref={ref}
-      {...other}
-    >
-      <ListContext.Provider value={context}>
+    <ListContext.Provider value={context}>
+      <Component
+        className={clsx(
+          classes.root,
+          {
+            [classes.dense]: dense,
```

Comment:
Why was it necessary to switch the provider and component position? It creates a huge diff around this tiny change.

Reply:
The main motivation is to match the other components. It's not necessary.